ethereum/contracts/relayer/wormholeRelayer/WormholeRelayerSend.sol

```diff
@@ -0,0 +1,541 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.19;
+
+import {
+    DeliveryProviderDoesNotSupportTargetChain,
+    InvalidMsgValue,
+    DeliveryProviderCannotReceivePayment,
+    VaaKey,
+    IWormholeRelayerSend
+} from "../../interfaces/relayer/IWormholeRelayerTyped.sol";
+import {IDeliveryProvider} from "../../interfaces/relayer/IDeliveryProviderTyped.sol";
+
+import {toWormholeFormat, fromWormholeFormat} from "../../libraries/relayer/Utils.sol";
+import {
+    DeliveryInstruction,
+    RedeliveryInstruction
+} from "../../libraries/relayer/RelayerInternalStructs.sol";
+import {WormholeRelayerSerde} from "./WormholeRelayerSerde.sol";
+import {ForwardInstruction, getDefaultDeliveryProviderState} from "./WormholeRelayerStorage.sol";
+import {WormholeRelayerBase} from "./WormholeRelayerBase.sol";
+import "../../interfaces/relayer/TypedUnits.sol";
+import "../../libraries/relayer/ExecutionParameters.sol";
+
+abstract contract WormholeRelayerSend is WormholeRelayerBase, IWormholeRelayerSend {
+    using WormholeRelayerSerde for *; 
+    using WeiLib for Wei;
+    using GasLib for Gas;
+    using TargetNativeLib for TargetNative;
+    using LocalNativeLib for LocalNative;
+
+    /*
+    * Public convenience overloads
+    */
+
+    function sendPayloadToEvm(
+        uint16 targetChain,
+        address targetAddress,
+        bytes memory payload,
```

Comment:
All readonly `bytes` arguments in external functions should be converted to `calldata` to save gas.

Reply:
I realized one reason we can't do this in some places is calldata takes up two stack slots while bytes just takes up one, and thats enough to take us into StackTooDeep territory :( 

We probably can only do this in a few places (in the two sendPayloadToEvm, and in forwardPayloadToEvm). will do it there!

Reply:
Ahh, I see.